iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/OutputDeclarationConverter.java

```diff
@@ -60,16 +60,16 @@ public OutputDeclaration convertOutputDeclaration(TupleTree tupleTree) {
     MappingTree outputMapping = toMappingTree(tupleTree.value());
     Map<String, Property> properties = extractProperties(outputMapping);
 
-    SimpleProperty type = extractMandatorySimpleProperty(tupleTree.metadata(), properties, "type");
-    SimpleProperty condition = extractSimpleProperty(properties, "condition");
-    SimpleProperty value = extractSimpleProperty(properties, "value");
-    SimpleProperty copyCount = null;
-    SimpleProperty copyInput = null;
+    Property type = extractMandatoryProperty(tupleTree.metadata(), properties, "type", ArmTree.Kind.STRING_LITERAL);
+    Property condition = extractProperty(properties, "condition", ArmTree.Kind.STRING_LITERAL);
+    Property value = extractProperty(properties, "value", ArmTree.Kind.STRING_LITERAL);
+    Property copyCount = null;
+    Property copyInput = null;
```

Comment:
This is working. However, you can simplify this logic by using `PropertyUtils.get(outputMapping, "condition").map(this::toStringLiteral).or(null);` or similar. Therefore you may also need to add the TextTree interface. So you want to be this to do in another ticket.